app/services/post_alerter.rb

```diff
@@ -393,7 +393,7 @@ def create_notification(user, type, post, opts = {})
       skip_send_email: skip_send_email
     )
 
-    if created.id && !existing_notification && NOTIFIABLE_TYPES.include?(type) && !user.suspended?
+    if created.id && !existing_notification && NOTIFIABLE_TYPES.include?(type) && !user.suspended? && !opts[:skip_push]
```

Comment:
If we only want to `skip_push`, it should only be applied to `push_notification(user, payload)`. This is currently disabling live updates to the app and the event trigger that we fire.

Reply:
That is the intended behaviour; I don't want the notifications to be serialized to the client or trigger any other side effects like sending push notifications or any other hooks baked into `post_notification_alert`. Maybe there's a better name?

Reply:
Can you provide us with a sample of how your plugin is using the post alerter? I'm more inclined in letter plugins customize the payload than having an option to disable "live" notification.

Reply:
Sure.

I want to create notifications in the database for people mentioned in a chat post. So, when someone types `Hello @tgxworld!`, a notification record is stored in the database, with your user ID, the person who mentioned you, and the post / topic that it occurred in. This is the set of info that I want, and makes it really easy to query for, for example, the number of unread mentions you have in your chats.

```
Notification.where(topic_id: <your chat topic ids>, read: false).count
```

I'd like to do this in the same way, that Discourse does for regular posts, which means calling `PostAlerter.after_save_post` on the chat post. This allows me to re-use the logic of pulling out username mentions, expanding group mentions, and all the other good stuff that happens when someone makes a post on Discourse.

However, I don't want that notification to appear in your notifications tray, because the notifications tray already has enough different kinds of notifications (normal, PM, and flags), the default behavior (linking to the post directly) is sensible but not what I want here and is something that's difficult to override both on the client and on the server.

So, essentially I'm looking for a way to call this code (post_alerter#386):
```
    # Create the notification
    created = user.notifications.create!(
      notification_type: type,
      topic_id: post.topic_id,
      post_number: post.post_number,
      post_action_id: opts[:post_action_id],
      data: notification_data.to_json,
      skip_send_email: skip_send_email
    )
```

without triggering side effects. However, it's in the heart of the long `create_notification` method, and there are ~70 lines of code which go into producing that `notification_data.to_json` bit that I'd like to avoid duplicating in my plugin.

Reply:
How can I help to resolve this? @tgxworld

Reply:
?? @tgxworld @SamSaffron

Reply:
I think what you need here is a custom notification type and not an option to skip creating a notification. From what you're describing, a "mention" in your chat plugin is not the same as a "mention" in the Discourse code base. As long as the type is not part of `PostAlerter::NOTIFIABLE_TYPES`, the server side wouldn't publish anything to the client.